src/control/lib/netdetect/netdetect.go

```diff
@@ -297,6 +297,29 @@ func getNodeSibling(deviceScanCfg DeviceScan) C.hwloc_obj_t {
 	return nil
 }
 
+// getNodeAlias finds a node object that is the sibling of the device being matched.
+// This function is less restrictive than getNodeSibling because it does not require that the search device
+// be a member of the systemDeviceNamesMap.
+func getNodeAlias(deviceScanCfg DeviceScan) C.hwloc_obj_t {
+	node := getNodeDirect(deviceScanCfg)
+	if node == nil || node.parent == nil {
+		return nil
+	}
+	// This node will have a sibling if its parent has more than one child (arity > 0)
+	// Search for the first sibling node that has a different name than the search node name.
+	if node.parent.arity > 0 {
+		count := C.uint(node.parent.arity)
+		log.Debugf("There are %d children of this parent node.", int(count))
+		children := (*[1 << 30]C.hwloc_obj_t)(unsafe.Pointer(node.parent.children))[:count:count]
```

Comment:
I have to admit I don't totally understand what's happening in this line. I can infer the intent, but the details are confusing. Does this mean the hwloc_obj_t array is 10^30 items long?

Reply:
This is a common convention in cgo for creating a Go slice backed by a C array: https://github.com/golang/go/wiki/cgo#turning-c-arrays-into-go-slices

Reply:
Ah, cool. You learn something new every day.